pkg/storage/storage.go

```diff
@@ -64,21 +63,41 @@ func CasList() []func(*client.K8sClient, []string) ([]metav1.TableColumnDefiniti
 func Describe(storages []string, openebsNs, casType string) error {
 	// 1. Create the clientset
 	k, _ := client.NewK8sClient(openebsNs)
-	// 2. Get the namespaces
+	// 2. Get the namespace
 	nsMap, _ := k.GetOpenEBSNamespaceMap()
 	if openebsNs == "" {
-		// TODO: Change this line, currently this overwriting empty flag values as cstor
-		if val, ok := nsMap[util.CstorCasType]; ok {
+		if casType == util.ZFSCasType {
+			// a temporary way to get the zfs-namespace
+			zfs, _, err := k.GetZFSNodes(nil, util.List, "", util.MapOptions{})
+			if err != nil {
+				return fmt.Errorf("please specify --openebs-namespace for ZFS LocalPV")
+			}
+			if zfs != nil && zfs.Items != nil && len(zfs.Items) > 0 {
+				k.Ns = zfs.Items[0].Namespace
+			}
+		} else if val, ok := nsMap[casType]; ok {
 			k.Ns = val
 		}
 	}
+	// 3. Run a specific cas-type function
+	if casType != "" {
+		if work, ok := CasDescribeMap()[casType]; ok {
+			for _, storage := range storages {
+				_ = work(k, storage)
+			}
+			return nil
+		}
+		return fmt.Errorf("cas-type %s unknown", casType)
+	}
+
+	// 4. Brute-force run describe the storage by all cas-type functions
 	for _, storageName := range storages {
-		// 3. Describe the storage
-		if list, ok := CasDescribeMap()[util.CstorCasType]; ok {
-			err := list(k, storageName)
-			if err != nil {
-				return err
+		for _, work := range CasDescribeList() {
+			if err := work(k, storageName); err == nil {
+				continue
```

Comment:
IMO. We should remove the continue and let the brute force logic describe all the pools having same name as the user has not specified any cas-type here.

For ex, I have one lvm pool named `awesomestorage` and a zfs pool also named `awesomestorage`.

Now the user enters, `kubectl openebs describe storage awesomestorage`, without specifying the cas type.
We should describe both the `awesomestorage`.

Reply:
The above would make more sense with the cas-type filtering we are providing.

Reply:
Makes sense, I'll drop the `continue`